JotunnLib/Entities/CustomRoom.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Jotunn.Configs;
+using Jotunn.Managers;
+using SoftReferenceableAssets;
+using UnityEngine;
+
+namespace Jotunn.Entities
+{
+    /// <summary>
+    ///     Main interface for adding custom dungeon rooms to the game.<br />
+    ///     All custom rooms have to be wrapped inside this class to add it to Jötunns <see cref="DungeonManager"/>.
+    /// </summary>
+    public class CustomRoom : CustomEntity
+    {
+        /// <summary>
+        ///     The prefab for this custom room.
+        /// </summary>
+        public GameObject Prefab { get; }
+
+        /// <summary>
+        ///     The <see cref="global::Room"/> component for this custom room as a shortcut.
+        /// </summary>
+        public Room Room { get; }
+
+        /// <summary>
+        ///     The name of this custom room as a shortcut.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        ///     Indicator if references from <see cref="Entities.Mock{T}"/>s will be replaced at runtime.
+        /// </summary>
+        public bool FixReference { get; set; }
+
+        /// <summary>
+        ///     Theme name of this room.
+        /// </summary>
+        public string ThemeName { get; private set; }
+
+        /// <summary>
+        ///     <see cref="Room.Theme"/> of this room.
+        /// </summary>
+        public Room.Theme Theme { get; private set; }
```

Comment:
Like in the original PR, this should be changed to primarily use the string ThemeName and convert to the `Room.Theme` internally if needed (can be cached).

At the very least, `Room.Theme Theme` should be made internal and not exposed to the API, a dev shouldn't have to worry about when to choose what or even theme name changes. Take a look at https://github.com/Valheim-Modding/Jotunn/blob/23cc2fd84ec72a1f19d17da4b8ed2f2b561af9e7/JotunnLib/Configs/CraftingStations.cs and https://github.com/Valheim-Modding/Jotunn/blob/22b24cba652698646b3ec3490e98ad00d30ad355/JotunnLib/Configs/PieceConfig.cs#L64 for example